src/libcollections/vec.rs

```diff
@@ -1301,6 +1306,47 @@ impl<T: Clone> Vec<T> {
     }
 }
 
+impl<T: Default> Vec<T> {
+    /// Resizes the `Vec` in-place so that `len` is equal to `new_len`.
+    ///
+    /// If `new_len` is greater than `len`, the `Vec` is extended by the
+    /// difference, with each additional slot filled with `Default::default()`.
+    /// If `new_len` is less than `len`, the `Vec` is simply truncated.
+    ///
+    /// This method uses `Default` to create new values on every push. If
+    /// you'd rather `Clone` a given value, use [`resize`].
+    ///
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// let mut vec = vec![1, 2, 3];
+    /// vec.resize_default(5);
+    /// assert_eq!(vec, [1, 2, 3, 0, 0]);
+    ///
+    /// let mut vec = vec![1, 2, 3, 4];
+    /// vec.resize_default(2);
+    /// assert_eq!(vec, [1, 2]);
+    /// ```
+    ///
+    /// [`resize`]: #method.resize
+    #[unstable(feature = "vec_resize_default", issue = "41758")]
+    pub fn resize_default(&mut self, new_len: usize) {
```

Comment:
I thought about that, but I decided we couldn't since that would require `T: Clone`, right? Same for calling `self.extend_with_element()`. Ideally we'd have some trait that is "produce more of the same value" which would be be implemented for both `T: Default` and `T: Clone`. Unless we have an impl `impl Clone for T where T: Default`, which could work (i.e., calling `Default::default` each time, but might be impossible to add today.